odoo/addons/base/models/assetsbundle.py

```diff
@@ -322,20 +373,75 @@ def save_attachment(self, type, content):
 
         return attachment
 
-    def js(self):
-        attachments = self.get_attachments('js')
-        if not attachments:
+    def js(self, is_minified=True):
+        extension = 'min.js' if is_minified else 'js'
+        attachments = self.get_attachments(extension, version_mode=VersionMode.INCLUDE if is_minified else VersionMode.NONE)
+
+        if attachments and (is_minified or self.last_modified < attachments[0].write_date):
+            return attachments[0]
+
+        if is_minified:
             content = ';\n'.join(asset.minify() for asset in self.javascripts)
-            return self.save_attachment('js', content)
-        return attachments[0]
+            return self.save_attachment(extension, content)
+        else:
+            return self.js_with_sourcemap()
+
+    def js_with_sourcemap(self):
+        """Create/modify the ir.attachment representing the not-minified content of the bundleJS
+        and the ir.attachment representing the linked sourcemap.
+        To avoid the loss of break points in the development tool when the bundleJS
+        has to be rebuilt, the bundleJS url must not change so we will not include
+        'self.version' in the ir.attachment url.
+        If any ir.attachment exists for the bundleJS, we create a new one (bundleJS and sourcemap),
+        else, we modify the 'raw' in the existing ir.attachments.
+
+        :return ir.attachment representing the un-minified content of the bundleJS
+        """
+        sourcemap_attachment = self.get_attachments('js.map', version_mode=VersionMode.NONE) \
+                                or self.save_attachment('js.map', '', with_version=False)
+        generator = SourceMapGenerator(
+            source_root="../../../",
+        )
+
+        content_bundle_list = []
+        content_line_count = 0
+        line_header = 6  # number of lines added by with_header()
+        for asset in self.javascripts:
+            if asset.is_transpiled:
+                # '+ 3' corresponds to the 3 lines added at the beginning of the file during transpilation.
+                generator.add_source(
+                    asset.url, asset._content, content_line_count, start_offset=line_header + 3)
```

Comment:
don't access private members
```suggestion
                    asset.url, asset.content, content_line_count, start_offset=line_header + 3)
```

Reply:
Oops, I realized that this is intensional.
C'est un peu dégueulasse, non?

Reply:
Un peu mais je t'avoue ne pas avoir de meilleur idée

Reply:
Alors laisse comme ça, on peut le fixer plus tard.